cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -663,7 +665,30 @@ func newStateMachine(disk disks.Disk) *stateMachine {
 func (m *stateMachine) execute() (finished bool, err error) {
 	next, err := m.current()
 	m.current = next
-	return next == nil, err
+	finished = next == nil
+	if finished && err == nil {
+		if err := m.finalize(); err != nil {
+			return true, err
+		}
+	}
+	return finished, err
+}
+
+func (m *stateMachine) finalize() error {
+	// check soundness
+	part := m.degradedState.partition("ubuntu-data")
+	if part.MountState == partitionMounted && m.isEncryptedDev {
+		// check that save and data match
+		trustData, _ := checkDataAndSavaPairing(boot.InitramfsHostWritableDir)
+		if !trustData {
+			part.MountState = partitionMountedUntrusted
```

Comment:
I think maybe we should add something to the error-log so we know that this happened